client/scripts/main.tsx

```diff
@@ -1,22 +1,51 @@
-import * as React from 'react';
-import { render, findDOMNode } from 'react-dom';
-import { createHistory } from 'history';
-import { Router, Route, History } from 'react-router';
+import * as React from "react";
+import { render, findDOMNode } from "react-dom";
+import { createHistory } from "history";
+import { Router, Route, History } from "react-router";
 
-import helpers from './helpers';
+import helpers from "./helpers";
 let h = new helpers();
 
 /**
  * Interfaces
  */
 interface HeaderProps {
-  tagline: string
+  tagline: string;
+}
+
+interface Fish {
+  name: string;
+  price: number;
+  status: string;
+  desc: string;
+  image: string;
+}
+
+interface AddFishProps {
+  /**
+   * takes an object of type Fish and saves it to the app state fishes
+   */
+  addFish(fish: Fish);
 }
 
 /**
  * App container
  */
 class App extends React.Component<any, any> {
+  constructor(props) {
```

Comment:
seeming I am using es6 classes and not createClass we need to use the constructor to setup the state.
https://facebook.github.io/react/docs/reusable-components.html#es6-classes